plugin/main/src/kotlinx/benchmark/gradle/NativeMultiplatformTasks.kt

```diff
@@ -116,30 +118,127 @@ fun Project.createNativeBenchmarkExecTask(
         onlyIf { linkTask.enabled }
 
         val reportsDir = benchmarkReportsDir(config, target)
-        val reportFile = reportsDir.resolve("${target.name}.json")
+        reportFile = reportsDir.resolve("${target.name}.json")
 
         val executableFile = linkTask.outputFile.get()
         executable = executableFile.absolutePath
-        if (target.workingDir != null)
-            workingDir = File(target.workingDir)
+        this.config = config
+        this.workingDir = target.workingDir?.let { File(it) }
 
         onlyIf { executableFile.exists() }
+        configDir = file(project.buildDir.resolve(target.extension.configsDir).resolve(config.name))
+        configDir.mkdirs()
+
+        val ideaActive = (extensions.extraProperties.get("idea.internal.test") as? String)?.toBoolean() ?: false
+        configFile = writeParameters(target.name, reportFile, if (ideaActive) "xml" else "text", config)
 
         dependsOn(linkTask)
         doFirst {
-            val ideaActive = (extensions.extraProperties.get("idea.internal.test") as? String)?.toBoolean() ?: false
-            args(writeParameters(target.name, reportFile, if (ideaActive) "xml" else "text", config))
             reportsDir.mkdirs()
             logger.lifecycle("Running '${config.name}' benchmarks for '${target.name}'")
         }
     }
 }
 
-open class NativeBenchmarkExec : Exec() {
+open class NativeBenchmarkExec() : DefaultTask() {
 /*
     @Option(option = "filter", description = "Configures the filter for benchmarks to run.")
     var filter: String? = null
 */
+    @Input
+    lateinit var executable: String
+
+    var workingDir: File? = null
+
+    @Input
+    lateinit var configFile: File
+
+    @Input
+    lateinit var config: BenchmarkConfiguration
+
+    @Input
+    lateinit var reportFile: File
+
+    @Input
+    lateinit var configDir: File
+
+    private fun execute(args: Collection<String>) {
+        project.exec {
+            it.executable = executable
+            it.args(args)
+            if (workingDir != null)
+                it.workingDir = workingDir
+        }
+    }
+
+    @ExperimentalPathApi
+    @TaskAction
+    fun run() {
+        // Get full list of running benchmarks
+        execute(listOf(configFile.absolutePath, "--list", configDir.absolutePath))
+        val detailedConfigFiles = project.fileTree(configDir).files.sortedBy { it.absolutePath }
+        val jsonReportParts = mutableListOf<File>()
+
+        detailedConfigFiles.forEach { runConfig ->
+            val runConfigPath = runConfig.absolutePath
+            val lines = runConfig.readLines()
+            require(lines.size > 1) { "Wrong detailed configuration format" }
+            val currentConfigDescription = lines[1]
+
+            // Execute benchmark
+            if (config.iterationMode == "internal") {
+                val jsonFile = createTempFile("bench", ".json").toFile()
```

Comment:
Just a suggestion: most `kotlin.io.*` functions have analogs in `kotlin.io.path`, so you can avoid converting the result of `createTempFile` to `File` and just use the returned `Path`.

Reply:
The problem is that `kotlin.io.createTempFile` is deprecated

```
@Deprecated(
    "Avoid creating temporary files in the default temp location with this function " +
    "due to too wide permissions on the newly created file. " +
    "Use kotlin.io.path.createTempFile instead or resort to java.io.File.createTempFile."
)

I used one of suggested alternatives

Reply:
Sure, I mean you can use the result of `kotlin.io.path.createTempFile` without converting it `toFile()` first. Though I don't insist.

Reply:
May be I didn't get. What was your suggestion? Use deprecated function in order to eliminate `toFile()`?  Or do you mean using `Path`, not `File`?